### PR TITLE
- Redo the servicepool to resolve addresses first time and not loadba…

### PR DESCRIPTION
**Pull request by baldersheim**
…lance.

- Make it thread safe.
- Remove any loadbalancing tests
- Assert that no loadbalancing is requested.

@vekterli PR

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1: some comments for consideration

**Comment by vekterli** [comment]
Also note failing CI test:
```
The following tests FAILED:
	208 - messagebus_test_error_test_app (Failed)
Errors while running CTest
```

File: messagebus/src/vespa/messagebus/network/rpcservice.cpp [hunk 1]
```diff
@@ -6,36 +6,36 @@
...
+            RPCServiceAddress test(pattern, pattern.substr(0, pos));
+            if ( ! test.isMalformed()) {
+                _serviceName = pattern;
+                _connectionSpec = pattern.substr(0, pos);
+            }
+        }
+    } else {
+        Mirror::SpecList addressList = mirror.lookup(pattern);
+        if (!addressList.empty()) {
+            assert(addressList.size() == 1);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/network/rpcservice.cpp]: That was my intention to leave it there for factory to punch me back if assumptions where wrong :) Added an urgent TODO.

File: messagebus/src/vespa/messagebus/network/rpcservicepool.cpp [hunk 2]
```diff
@@ -19,27 +22,52 @@ RPCServicePool::~RPCServicePool() = default;
...
+        LockGuard guard(_lock);
+        handleMirrorUpdates(guard);
+        std::shared_ptr<RPCService> *found = _lru->findAndRef(pattern);
+        if (found) {
+            service = *found;
+        }
+    }
+
+    if (service) {
+        return service->resolve();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/network/rpcservicepool.cpp]: Agree, fixed

File: messagebus/src/vespa/messagebus/network/rpcservice.cpp [hunk 3]
```diff
@@ -6,36 +6,36 @@
...
+            const auto &entry = addressList[random() % addressList.size()];
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/network/rpcservice.cpp]: Done

**Comment by baldersheim** [comment]
I am not able to reproduce locally

error.cpp:31: error: check failure #1: '0 == system("./messagebus_test_cpp-client-error_app")' in thread '0(1)' (error.cpp)
    STATE[0]: 'error_test' (error.cpp:15)
lhs: 0
rhs: 139

Tried on 2 machines, both centos and rhel6.

**Pull request merged**